internal/receiver/discoveryreceiver/README.md

```diff
@@ -0,0 +1,334 @@
+configured to use the `collectd/redis` monitor. The `status` mapping comprises entries that signal the Smart Agent
+receiver has been instantiated with a `successful`, `partial`, or `failed` status, based on reported `metrics` or
+recorded application log `statements`.
+
+The Redis receiver will warrant a status log record from the Discovery receiver of...
+
+* `successful` if it emits any metrics matching the `regexp: .*` pattern, denoting that metric gathering and the
+Receiver are functional.
+* `partial` if it internally logs a statement matching the `regexp: (WRONGPASS|NOAUTH|ERR AUTH)` pattern, suggesting
+there is a Redis server but it's receiving incorrect credentials.
+* `failed` if it internally logs a statement matching the `regexp: ConnectionRefusedError` pattern, suggesting that no
+Redis server is available at the endpoint.
+
+```yaml
+extensions:
+  docker_observer:
+receivers:
+   discovery:
+     watch_observers: [docker_observer]
+     receivers:
+       smartagent/redis:
+         // Determine the functionality of the smartagent/redis receiver for any detected container
+         rule: type == "container"
+         config:
+           type: collectd/redis
+           // the metric or log statement
+           status:
+             metrics:
+               successful:
+                 - regexp: '.*'
+                   // Only emit a single log record for this status entry instead of one for each matching received metric (`false`, the default)
+                   first_only: true
+                   record:
+                     severity: info
+                     body: Successfully able to connect to Redis container.
+             statements:
+               partial:
+                 - regexp: (WRONGPASS|NOAUTH|ERR AUTH)
+                   first_only: true
+                   record:
+                     severity: warn
+                     body: Container appears to be accepting redis connections but the default auth setting is incorrect. 
+               failed:
+                 - regexp: ConnectionRefusedError
+                   first_only: true
+                   record:
+                     severity: debug
+                     body: Container appears to not be accepting redis connections.
+exporters:
+  logging:
+    logLevel: debug
+    sampling_initial: 1
+    sampling_thereafter: 1
+service:
+  extensions:
+    - docker_observer
+  pipelines:
+    logs:
+      receivers:
+        - discovery
+      exporters:
+        - logging
+```
+
+Given this configuration, if the Discovery receiver's Docker observer instance reports an active Redis container and
+the Smart Agent receiver's associated `collectd/redis` monitor is able to generate metrics for the container, something
+like the following log record would be emitted by the receiver:
+
```

Comment:
Suggestion:

Given this configuration, if the Discovery receiver's Docker observer instance reports an active Redis container, and the Smart Agent receiver's associated `collectd/redis` monitor is able to generate metrics for the container, the receiver will emit something similar to the following log record:

Reply:
I don't know why, but I'm a fan of passive voice (updated).